### PR TITLE
Revert "Better support new assembly"

### PR DESCRIPTION
**Pull request by thomasmaurel**
Reverts Ensembl/ensembl-prodinf-core#29.

I will create a new PR for making the code a bit more pythony.

### CODE REVIEW

**Comment by thomasmaurel** [comment]
This code should be done in the metadata updater perl API and not in the handover task

**Comment by thomasmaurel** [comment]
Is this ok @marcoooo ?

File: ensembl_prodinf/handover_tasks.py [hunk 1]
```diff
@@ -73,21 +71,18 @@ def handover_database(spec):
...
         raise ValueError(spec['src_uri'] + " has been handed over after the deadline. Please contact the Production team")
-    #Check if this is a new assembly
-    spec = check_new_assembly(spec,cfg.metadata_uri,db_prefix,release,db_type)
     #Get database hc group and compara_uri
     (groups,compara_uri) = hc_groups(db_type,db_prefix,spec['src_uri'])
     #Check to which staging server the database need to be copied to
     (spec,staging_uri) = check_staging_server(spec,db_type,db_prefix,assembly)
     #setting compara url to default value for species databases. This value is only used by Compara healthchecks
-    if compara_uri is None:
+    if (compara_uri == None):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if compara_uri is None:
```

File: ensembl_prodinf/handover_tasks.py [hunk 2]
```diff
@@ -100,25 +95,25 @@ def get_tgt_uri(src_url,staging_uri):
     
 def check_db(uri):    
     """Check if source database exists"""
-    if not database_exists(uri):
+    if(database_exists(uri) == False):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if not database_exists(uri):
```

File: ensembl_prodinf/handover_tasks.py [hunk 3]
```diff
@@ -100,25 +95,25 @@ def get_tgt_uri(src_url,staging_uri):
...
         get_logger().error("Handover failed, " + uri + " does not exist")
         raise ValueError(uri + " does not exist")
     else:
         return
 
 def parse_db_infos(database):
     """Parse database name and extract db_prefix and db_type. Also extract release and assembly for species databases"""
-    if species_pattern.match(database):
+    if(species_pattern.match(database)):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if species_pattern.match(database):
```

File: ensembl_prodinf/handover_tasks.py [hunk 4]
```diff
@@ -100,25 +95,25 @@ def get_tgt_uri(src_url,staging_uri):
...
         db_prefix = species_pattern.match(database).group(1)
         db_type = species_pattern.match(database).group(2)
         release = species_pattern.match(database).group(3)
         assembly = species_pattern.match(database).group(4)
         return db_prefix, db_type, release, assembly
-    elif compara_pattern.match(database):
+    elif(compara_pattern.match(database)):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
     elif compara_pattern.match(database):
```

File: ensembl_prodinf/handover_tasks.py [hunk 5]
```diff
@@ -100,25 +95,25 @@ def get_tgt_uri(src_url,staging_uri):
...
         db_type = compara_pattern.match(database).group(1)
         db_prefix = compara_pattern.match(database).group(2)
         return db_prefix, db_type, None, None
-    elif ancestral_pattern.match(database):
+    elif(ancestral_pattern.match(database)):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif ancestral_pattern.match(database):
```

File: ensembl_prodinf/handover_tasks.py [hunk 6]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
 
 def hc_groups(db_type,db_prefix,uri):
     """Find which HC group to run on a given database type. For Compara generate the compara master uri"""
-    if db_type in ['core','rnaseq','cdna','otherfeatures']:
+    if(db_type in ['core','rnaseq','cdna','otherfeatures']):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if db_type in ['core','rnaseq','cdna','otherfeatures']:
```

File: ensembl_prodinf/handover_tasks.py [hunk 7]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
         return [cfg.core_handover_group],None
-    elif db_type == 'variation':
+    elif(db_type == 'variation'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif db_type == 'variation':
```

File: ensembl_prodinf/handover_tasks.py [hunk 8]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
         return [cfg.variation_handover_group],None
-    elif db_type == 'funcgen':
+    elif(db_type == 'funcgen'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif db_type == 'funcgen':
```

File: ensembl_prodinf/handover_tasks.py [hunk 9]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
         return [cfg.funcgen_handover_group],None
-    elif db_type == 'ancestral':
+    elif(db_type == 'ancestral'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif db_type == 'ancestral':
```

File: ensembl_prodinf/handover_tasks.py [hunk 10]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
         return [cfg.ancestral_handover_group],None
-    elif db_type == 'compara':
-        if  db_prefix == "pan":
+    elif(db_type == 'compara'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif db_type == 'compara':
```

Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif db_type == 'compara':
```

File: ensembl_prodinf/handover_tasks.py [hunk 11]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
+        if (db_prefix == "pan"):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
        if db_prefix == "pan":
```

File: ensembl_prodinf/handover_tasks.py [hunk 12]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
             compara_uri=cfg.compara_uri + db_prefix + '_compara_master'
             compara_handover_group=cfg.compara_pan_handover_group
-        elif check_grch37(uri,'homo_sapiens'):
+        elif (check_grch37(uri,'homo_sapiens')):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
        elif check_grch37(uri,'homo_sapiens'):
```

File: ensembl_prodinf/handover_tasks.py [hunk 13]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
             compara_uri=cfg.compara_uri + 'ensembl_compara_master_grch37'
             compara_handover_group=cfg.compara_handover_group
-        elif db_prefix:
+        elif (db_prefix):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
        elif db_prefix:
```

File: ensembl_prodinf/handover_tasks.py [hunk 14]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
             compara_uri=cfg.compara_uri + 'ensembl_compara_master'
             compara_handover_group=cfg.compara_handover_group
         return [compara_handover_group],compara_uri
-    else:
-        raise ValueError("Can't find hc group for "+uri+" Please check handover config file")
 
 def check_staging_server(spec,db_type,db_prefix,assembly):
     """Find which staging server should be use. secondary_staging for GRCh37 and Bacteria, staging for the rest"""
-    if 'bacteria' in db_prefix:
+    if('bacteria' in db_prefix):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if 'bacteria' in db_prefix:
```

File: ensembl_prodinf/handover_tasks.py [hunk 15]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
         staging_uri = cfg.secondary_staging_uri
-    elif db_prefix == 'homo_sapiens' and assembly == '37':
+    elif(db_prefix == 'homo_sapiens' and assembly == '37'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif db_prefix == 'homo_sapiens' and assembly == '37':
```

File: ensembl_prodinf/handover_tasks.py [hunk 16]
```diff
@@ -127,66 +122,45 @@ def parse_db_infos(database):
...
         staging_uri = cfg.secondary_staging_uri
         spec['GRCh37']=1
         spec['progress_total']=2
-    elif db_type == 'compara' and check_grch37(spec['src_uri'],'homo_sapiens'):
+    elif(db_type == 'compara' and check_grch37(spec['src_uri'],'homo_sapiens')):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif db_type == 'compara' and check_grch37(spec['src_uri'],'homo_sapiens'):
```

File: ensembl_prodinf/handover_tasks.py [hunk 17]
```diff
@@ -207,20 +181,20 @@ def process_checked_db(self, hc_job_id, spec):
     self.max_retries = None
     get_logger().info("HCs in progress, please see: " +cfg.hc_web_uri + str(hc_job_id))
     result = hc_client.retrieve_job(hc_job_id)
-    if result['status'] in ['incomplete', 'running', 'submitted']:
+    if (result['status'] == 'incomplete') or (result['status'] == 'running') or (result['status'] == 'submitted'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] in ['incomplete', 'running', 'submitted']:
```

File: ensembl_prodinf/handover_tasks.py [hunk 18]
```diff
@@ -207,20 +181,20 @@ def process_checked_db(self, hc_job_id, spec):
...
         get_logger().debug("HC Job incomplete, checking again later")
         raise self.retry()
     
     # check results
-    if result['status'] == 'failed':
+    if (result['status'] == 'failed'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] == 'failed':
```

File: ensembl_prodinf/handover_tasks.py [hunk 19]
```diff
@@ -207,20 +181,20 @@ def process_checked_db(self, hc_job_id, spec):
...
         get_logger().info("HCs failed to run, please see: "+cfg.hc_web_uri + str(hc_job_id))
         msg = """
 Running healthchecks on %s failed to execute.
 Please see %s
 """ % (spec['src_uri'], cfg.hc_web_uri + str(hc_job_id))
         send_email(to_address=spec['contact'], subject='HC failed to run', body=msg, smtp_server=cfg.smtp_server)
         return 
-    elif result['output']['status'] == 'failed':
+    elif (result['output']['status'] == 'failed'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif result['output']['status'] == 'failed':
```

File: ensembl_prodinf/handover_tasks.py [hunk 20]
```diff
@@ -263,18 +227,18 @@ def process_copied_db(self, copy_job_id, spec):
     self.max_retries = None
     get_logger().info("Copying in progress, please see: " +cfg.copy_web_uri + str(copy_job_id))
     result = db_copy_client.retrieve_job(copy_job_id)
-    if result['status'] in ['incomplete', 'running', 'submitted']:
+    if (result['status'] == 'incomplete') or (result['status'] == 'running') or (result['status'] == 'submitted'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] in ['incomplete', 'running', 'submitted']:
```

File: ensembl_prodinf/handover_tasks.py [hunk 21]
```diff
@@ -263,18 +227,18 @@ def process_copied_db(self, copy_job_id, spec):
...
         get_logger().debug("Database copy job incomplete, checking again later")
         raise self.retry()
-    if result['status'] == 'failed':
+    if (result['status'] == 'failed'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] == 'failed':
```

File: ensembl_prodinf/handover_tasks.py [hunk 22]
```diff
@@ -263,18 +227,18 @@ def process_copied_db(self, copy_job_id, spec):
...
         get_logger().info("Copy failed, please see: "+cfg.copy_web_uri + str(copy_job_id))
         msg = """
 Copying %s to %s failed.
 Please see %s
 """ % (spec['src_uri'], spec['tgt_uri'], cfg.copy_web_uri + str(copy_job_id))
         send_email(to_address=spec['contact'], subject='Database copy failed', body=msg, smtp_server=cfg.smtp_server)
         return
-    elif 'GRCh37'in spec:
+    elif('GRCh37'in spec):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif 'GRCh37'in spec:
```

File: ensembl_prodinf/handover_tasks.py [hunk 23]
```diff
@@ -301,10 +265,10 @@ def process_db_metadata(self, metadata_job_id, spec):
     self.max_retries = None
     get_logger().info("Loading into metadata database, please see: "+cfg.meta_uri + "jobs/"+ str(metadata_job_id))
     result = metadata_client.retrieve_job(metadata_job_id)
-    if result['status'] in ['incomplete', 'running', 'submitted']:
+    if (result['status'] == 'incomplete') or (result['status'] == 'running') or (result['status'] == 'submitted'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] in ['incomplete', 'running', 'submitted']:
```

File: ensembl_prodinf/handover_tasks.py [hunk 24]
```diff
@@ -301,10 +265,10 @@ def process_db_metadata(self, metadata_job_id, spec):
...
         get_logger().debug("Metadata load Job incomplete, checking again later")
         raise self.retry()
-    if result['status'] == 'failed':
+    if (result['status'] == 'failed'):
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] == 'failed':
```

**Review by marcoooo — CHANGES_REQUESTED** [review]

File: ensembl_prodinf/handover_tasks.py (hunk 19 above)
Comment (thomasmaurel) [reply ensembl_prodinf/handover_tasks.py]: Making code more Pythony

**Comment by thomasmaurel** [comment]
Ok to merge now @marcoooo ?

File: ensembl_prodinf/handover_tasks.py (hunk 4 above)
Comment (thomasmaurel) [reply ensembl_prodinf/handover_tasks.py]: @marcoooo: You added space by mistake here. This will break the whole file.

Comment (thomasmaurel) [reply ensembl_prodinf/handover_tasks.py]:
```suggestion
   elif(compara_pattern.match(database)):
```

Comment (thomasmaurel) [reply ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif compara_pattern.match(database):
```

File: ensembl_prodinf/handover_tasks.py [hunk 25]
```diff
@@ -114,7 +109,7 @@ def parse_db_infos(database):
         release = species_pattern.match(database).group(3)
         assembly = species_pattern.match(database).group(4)
         return db_prefix, db_type, release, assembly
-    elif compara_pattern.match(database):
+     elif compara_pattern.match(database):
```
Comment (thomasmaurel) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    elif compara_pattern.match(database):
```

File: ensembl_prodinf/handover_tasks.py (hunk 4 above)
Comment (thomasmaurel) [reply ensembl_prodinf/handover_tasks.py]: Ok, I will create a fix for this @marcoooo

**Comment by thomasmaurel** [comment]
@marcoooo: I think this can be merged now.

**Review by marcoooo — APPROVED** [review]

**Pull request merged**